codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientClass.java

```diff
@@ -154,6 +162,17 @@ private MethodSpec constructor() {
                                  EndpointDiscoveryRefreshCache.class,
                                  poetExtensions.getClientClass(model.getNamingStrategy().getServiceName() +
                                                                "EndpointDiscoveryCacheLoader"));
+
+            if (model.getCustomizationConfig().allowEndpointOverrideForEndpointDiscoveryRequiredOperations()) {
+                builder.beginControlFlow("if (clientConfiguration.option(SdkClientOption.ENDPOINT_OVERRIDDEN) == "
+                                         + "Boolean.TRUE)");
+                builder.addStatement("log.warn(() -> $S)",
+                                     "Endpoint discovery is enabled for this client, and an endpoint override was also "
+                                     + "specified. This will disable endpoint discovery for methods that require it, instead "
```

Comment:
`if (clientConfiguration.option(SdkClientOption.ENDPOINT_DISCOVERY_ENABLED))` is specified above on line 160, so we know that endpoint discovery is enabled here.

Reply:
I didn't see before that the new if statement is within the flow of the ENDPOINT_DISCOVERY_ENABLED codegen-if, which it is. So that's fine, it's 
endpoint discovery operation AND ENDPOINT_DISCOVERY_ENABLED and ENDPOINT_OVERRIDDEN allowEndpointOverrideForEndpointDiscoveryRequiredOperations which will trigger the log warning. Many ifs :-(